src/components/editor/editor-window/tool-bar/utils.ts

```diff
@@ -0,0 +1,128 @@
+import CodeMirror from 'codemirror'
+
```

Comment:
Did you know there is an array function `splice` that can delete items from an array AND insert items into the array at specific indexes? This and the usage of map and filter could make the content a lot easier to understand.

Reply:
I stoped looking through this file after the `j = i` comment. Please clean this up first. :/

Reply:
I just have a look at the splice functions and as far as the documentation tells me this helps with nothing at all

Reply:
I used splice in one section where I could use it